### PR TITLE
[MMCA-3612] External view of standing authority following the introduction of an XI EORI (DDAs, GGAs, cash account) - find account you have authority to use - using account id(Frontend changes 1)

### PR DESCRIPTION
**Pull request by tyagikumarvineet**
Description - Changes have been done for account id as a search query while considering the XI related authorities

Below have been done as part of this PR

- Tests cases for controller and new test classes for various views have been added
- Code changes in AuthorizedToViewController
- refactoring
- Addition of a new Utils class

### CODE REVIEW

File: app/views/components/authorised_account_search_company_name.scala.html [hunk 1]
```diff
@@ -36,19 +41,38 @@
...
-  </div>
- }
-
- <div class="govuk-summary-list__row">
-  <dt class="govuk-summary-list__key govuk-!-width-three-quarters govuk-!-padding-top-0 govuk-!-padding-bottom-0">
-   @messages("cf.search.authorities.result.eori.number")</dt>
-  <dd class="govuk-!-width-three-quarters govuk-summary-list__value govuk-!-padding-top-0 govuk-!-padding-bottom-0">@query</dd>
- </div>
+    @companyName.map { name =>
+    <div class="govuk-summary-list__row">
```
Comment (wg-hmrc) [inline app/views/components/authorised_account_search_company_name.scala.html:45]: Not a biggie however we should try avoid using HTML syntax and import the GOV.UK components, there is already a component for this https://design-system.service.gov.uk/components/summary-list/

File: app/controllers/AuthorizedToViewController.scala [hunk 2]
```diff
@@ -69,56 +69,180 @@ class AuthorizedToViewController @Inject()(authenticate: IdentifierAction,
...
-            Future.successful(BadRequest(authorisedToViewSearch(form.withError("value",
-              "cf.account.authorized-to-view.search-own-accountnumber").fill(query),
-                Some(""), LocalDate.now.toString, false)))
+      query => processSearchQuery(request, query)
+    )
+  }
+
+  private def processSearchQuery(request: AuthenticatedRequest[AnyContent],
+                                 query: EORI)(
+                                  implicit hc: HeaderCarrier,
```
Comment (AdamCoops) [inline app/controllers/AuthorizedToViewController.scala:90]: this spacing looks slightly indented here?

File: app/controllers/AuthorizedToViewController.scala [hunk 3]
```diff
@@ -69,56 +69,180 @@ class AuthorizedToViewController @Inject()(authenticate: IdentifierAction,
...
+      }
+    }
+    result.flatten
+  }
+
+  /**
+   * Search and processes the Authorities for the valid input, further, displays the view accordingly
+   */
+  private def searchAuthoritiesForValidInput(request: AuthenticatedRequest[AnyContent],
+                                              searchQuery: EORI,
```
Comment (AdamCoops) [inline app/controllers/AuthorizedToViewController.scala:130]: same here

File: app/views/authorised_to_view/authorised_to_view_search.scala.html [hunk 4]
```diff
@@ -32,7 +32,12 @@
         button: components.button
 )
 
-@(form: Form[_], url: Option[String], date: String, fileExists: Boolean)(implicit request: Request[_], messages: Messages, appConfig: AppConfig)
+@(
+form: Form[_],
```
Comment (AdamCoops) [inline app/views/authorised_to_view/authorised_to_view_search.scala.html:36]: Whats our thinking for have each param with 0 indentation on the left edge? I can see we're doing a few times here.

File: app/controllers/AuthorizedToViewController.scala (hunk 2 above)
Comment (tyagikumarvineet) [reply app/controllers/AuthorizedToViewController.scala:90]: I crossed check and its displaying fine for me. Hence resolving

File: app/controllers/AuthorizedToViewController.scala (hunk 3 above)
Comment (tyagikumarvineet) [reply app/controllers/AuthorizedToViewController.scala]: indentation for the method name was fine but I reformatted the method parameters.Have a look

File: app/views/authorised_to_view/authorised_to_view_search.scala.html (hunk 4 above)
Comment (tyagikumarvineet) [reply app/views/authorised_to_view/authorised_to_view_search.scala.html]: The only idea was to keep parameters on the separate lines in order to make these easily readable as it is very difficult to read all the parameters are on a single line. We have done the same for this constructor for the views. Indentation has been added now. Have a look and let me know if that's okay

**Review by AdamCoops — APPROVED** [review]

File: app/views/authorised_to_view/authorised_to_view_search.scala.html (hunk 4 above)
Comment (tyagikumarvineet) [reply app/views/authorised_to_view/authorised_to_view_search.scala.html]: Reviewer is happy with the change. Hence resolving.

File: app/controllers/AuthorizedToViewController.scala (hunk 3 above)
Comment (tyagikumarvineet) [reply app/controllers/AuthorizedToViewController.scala]: Reviewer is happy with the change. Hence resolving.

**Pull request merged**